datafusion/expr-common/src/signature.rs

```diff
@@ -225,6 +229,45 @@ impl Display for TypeSignatureClass {
     }
 }
 
+impl TypeSignatureClass {
+    /// Returns the default cast type for the given `TypeSignatureClass`.  
```

Comment:
I am not familiar what a "default cast type" is

I looked at the definition of LogicalType:defaut_cast_for and that didn't help me either (no comments)
https://github.com/apache/datafusion/blob/08d3b656d761a8e0d9e7f7c35c7b76cfec00e095/datafusion/common/src/types/native.rs#L201-L200

🤔